assets/differentialabundance_report.Rmd

```diff
@@ -1003,6 +1003,69 @@ for (i in 1:nrow(contrasts)){
             contrast_de <- round_dataframe_columns(contrast_de, digits=params$round_digits)
             print( htmltools::tagList(datatable(contrast_de, caption = paste('Differential genes', dir, 'in', contrast_descriptions[i], " (check", differential_files[[i]], "for more detail)"), rownames = FALSE) ))
 
+        # Boxplots for Deseq2
+        for (assay_type in rev(names(assay_data))){
+            cat(paste0("\n###### ", prettifyVariablename(assay_type), "\n"))
+            expr_mat <- assay_data[[assay_type]]
+
+            qval_column <- prettifyVariablename(params$differential_qval_column)
+
+            top_genes <- head(contrast_de[order(contrast_de[,qval_column]), 1], 10)
+            top_genes <- as.character(top_genes)
+
+            # # Subset observations to only samples in expr_mat
+            obs_sub <- observations[colnames(expr_mat), , drop = FALSE]
+
+            # library(reshape2)
+            plot_df <- reshape2::melt(expr_mat[top_genes, ])
```

Comment:
I think `tidyr::pivot_longer` is the more modern solution, and it doesn't require you an extra package (pretty sure tidyr is already loaded)